transifex/pull.py

```diff
@@ -6,238 +6,44 @@
-
-    def update_translations(self):
-        """Download and compile messages from Transifex.
-
-        Assumes this repo defines the `pull_translations` Make target and a
-        project config file at .tx/config. Running the Transifex client also
-        requires specifying Transifex credentials at ~/.transifexrc.
-
-        See http://docs.transifex.com/client/config/.
-        """
-        subprocess.run(['make', 'pull_translations'], check=True)
-
-        if self.skip_compilemessages:
-            logger.info('Skipping compilemessages.')
-            return
-
-        # Messages may fail to compile (e.g., a translator may accidentally translate a
-        # variable in a Python format string). If this happens, we want to proceed with
-        # the PR process and notify the team that messages failed to compile.
-        try:
-            # The compilemessages command is a script that (as of Django 1.9) scans the project
-            # tree for .po files to compile and calls GNU gettext's msgfmt command on them. It
-            # doesn't require DJANGO_SETTINGS_MODULE to be defined when run from the project root,
-            # and also doesn't need django.setup() to be run. Because of this, we can get away with
-            # django-admin.py instead of manage.py. The latter defines a default value for
-            # DJANGO_SETTINGS_MODULE and causes django.setup() to run, which is undesirable here for reasons
-            # ranging from Python 2/3 incompatibility errors across projects to forcing the installation
-            # of packages which provide installed apps custom to each project.
-            subprocess.run(['django-admin.py', 'compilemessages'], check=True)
-        except subprocess.CalledProcessError:
-            self.compilemessages_failed = True
-
-    def is_changed(self):
-        """Determine whether any changes were made."""
-        completed_process = subprocess.run(['git', 'status', '--porcelain'], stdout=subprocess.PIPE, check=True)
-        return bool(completed_process.stdout)
-
-    def commit(self):
-        """Commit changes.
-
-        Adds any untracked files, in case new translations are added.
-        """
-        subprocess.run(['git', 'add', '-A'], check=True)
-        try:
-            subprocess.run(['git', 'commit', '-m', self.message], check=True)
-        except subprocess.CalledProcessError:
-            subprocess.run(
-                [
-                    'git',
-                    '-c', 'user.name="{}"'.format(os.environ['GIT_USER_NAME']),
-                    '-c', 'user.email={}'.format(os.environ['GIT_USER_EMAIL']),
-                    'commit', '-m', self.message
-                ],
-                check=True
-            )
 
-    def push(self):
-        """Push branch to the remote."""
-        subprocess.run(['git', 'push', '-u', 'origin', self.branch_name], check=True)
+from utils import logger, repo_context
```

Comment:
It looks like Python is smart enough not to require this, but I tend to write fully qualify these import paths for clarity. In this case, that would be `from transifex.utils import ...`.